src/utils/bisect.js

```diff
@@ -0,0 +1,72 @@
+// This file originally comes from https://github.com/3rd-Eden/node-bisection/ but has been
+// imported and changed to fix some bugs.
+
+/*
+    Copyright (c) 2010/2011 Arnout Kazemier,3rd-Eden
+    Permission is hereby granted, free of charge, to any person obtaining a copy
+    of this software and associated documentation files (the "Software"), to deal
+    in the Software without restriction, including without limitation the rights
+    to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+    copies of the Software, and to permit persons to whom the Software is
+    furnished to do so, subject to the following conditions:
+    The above copyright notice and this permission notice shall be included in
+    all copies or substantial portions of the Software.
+    THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+    IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+    FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+    AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+    LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+    OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+    THE SOFTWARE. 
+*/
+
+/*
+    The two functions here, bisectionLeft and bisectionRight, return the index where a new element
+    would be inserted, respectively at the left or at the right of elements with the same value. 
+*/
+
+// @flow
+
+export function bisectionLeft(
+  array: number[] | ArrayBufferView,
```

Comment:
Try with `$ArrayBufferView` instead of ArrayBufferView (this is a Flow thing, because ArrayBufferView itself doesn't exist, but they needed a type to represent it. See https://github.com/facebook/flow/blob/2c36afede67b1fc71e7a76695e1f7988f3545c1d/lib/core.js#L1914 for the definition)

Reply:
Made this change, but the flow checks are still failing. I did try a couple of different things locally but couldn't get those to work. Basically I believe, length and index are not part of the DataView, but maybe if this were converted to an Int32Array or maybe a Uint32Array, it would work. But I'm fairly new to this so I will read a little more, and meanwhile if you have a hint of how we can go about fixing this, please let me know.

Reply:
Yeah, `$ArrayBufferView` doesn't seem like the right think, rather `$TypedArray` should be better (you can see it on the same line I linked above)

Reply:
Yes, that worked!